src/utils/mtev_memory.c

```diff
@@ -69,7 +74,6 @@ static __thread int begin_end_depth = 0;
 static __thread uint64_t needs_maintenance = 0;
 static void *mtev_memory_gc(void *unused);
 static mtev_log_stream_t mem_debug = NULL;
-static pthread_mutex_t mem_debug_lock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Why was `mem_debug` lazy loaded with a lock before but not needed now ?

Reply:
Long ago, the log find didn't do an implicit create. Now, it is much more sane